packages/redux/src/subscriptions/actions/factories/fetchUserSubscriptionsFactory.ts

```diff
@@ -0,0 +1,51 @@
+import * as actionTypes from '../../actionTypes';
+import type { Dispatch } from 'redux';
+import type { FetchUserSubscriptionsAction } from '../../types';
+import type { FetchUserSubscriptionsFactory } from './types';
+import type {
+  GetSubscriptions,
+  Subscription,
+} from '@farfetch/blackout-client/subscriptions/types';
+
+/* eslint-disable jsdoc/no-undefined-types */
+
+/**
+ * Method to create a thunk factory configured with the specified client.
+ *
+ * @function fetchUserSubscriptionsFactory
+ * @memberof module:subscriptions/actions/factories
+ *
+ * @param getSubscriptions - Get subscriptions client.
+ *
+ * @returns Thunk factory.
+ */
+const fetchUserSubscriptionsFactory: FetchUserSubscriptionsFactory<
+  GetSubscriptions
+> =
+  getSubscriptions =>
+  (query, config) =>
+  async (
+    dispatch: Dispatch<FetchUserSubscriptionsAction>,
+  ): Promise<Subscription[]> => {
```

Comment:
No need to type the return type as well as it is already set by the FetchUserSubscriptionsFactory <GetSubscriptions> type.